curvefs/src/mds/chunkid_allocator.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+/*
+ * Project: curve
+ * Created Date: 2021-08-27
+ * Author: chengyi01
+ */
+
+#include "curvefs/src/mds/chunkid_allocator.h"
+
+namespace curvefs {
+namespace mds {
+
+int ChunkIdAllocatorImpl::GenChunkId(uint64_t* chunkId) {
+    int ret = 0;
+    // will unloack at destructor
+    ::curve::common::WriteLockGuard guard(nextIdRWlock_);
+
+    if (nextId_ > lastId_ || CHUNKIDINITIALIZE == nextId_) {
+        // the chunkIds is exhausted || first to get chunkId
+        ret = AllocateBundleIds(bundleSize_);
```

Comment:
if ret is abnormal， why process in the if statement？

Reply:
the process deal with normal and allocate chunkid is same so process out of the  if statement